R-package/vignettes/CharRnnModel.Rmd

```diff
@@ -0,0 +1,277 @@
+Char RNN Example
+=============================================
+
+This example aims to show how to use lstm model to build a char level language model, and generate text from it. We use a tiny shakespeare text for demo purpose.
+
+Data can be found at https://github.com/dmlc/web-data/tree/master/mxnet/tinyshakespeare. 
+
+Preface
+-------
+This tutorial is written in Rmarkdown.
+- You can directly view the hosted version of the tutorial from [MXNet R Document](http://mxnet.readthedocs.org/en/latest/package/r/CharRnnModel.html)
+- You can find the download the Rmarkdown source from [here](https://github.com/dmlc/mxnet/blob/master/R-package/vignettes/CharRnnModel.Rmd)
+
+Load Data 
+---------
+First of all, load in the data and preprocess it.
+```{r}
+require(mxnet)
+```
+Set basic network parameters.
+```{r}
+batch.size = 32
+seq.len = 32
+num.hidden = 256
+num.embed = 256
+num.lstm.layer = 2
+num.round = 5
+learning.rate= 0.1
+wd=0.00001
+clip_gradient=1
+update.period = 1
+```
+download the data.
+```{r}
+download.data <- function(data_dir) {
+    dir.create(data_dir, showWarnings = FALSE)
+    setwd(data_dir)
+    if (!file.exists('input.txt')) {
+        download.file(url='https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/tinyshakespeare/input.txt',
+                      destfile='input.txt', method='wget')
+    }
+    setwd("..")
```

Comment:
I will merge this if all tests pass.

On Mon, Jun 6, 2016 at 9:47 AM, Yuqi Li <notifications@github.com> wrote:

> In R-package/vignettes/CharRnnModel.Rmd
> <https://github.com/dmlc/mxnet/pull/2329#discussion_r65903056>:
>
> > +num.round = 5
> > +learning.rate= 0.1
> > +wd=0.00001
> > +clip_gradient=1
> > +update.period = 1
> > +```
> > +download the data.
> > +```{r}
> > +download.data <- function(data_dir) {
> > +    dir.create(data_dir, showWarnings = FALSE)
> > +    setwd(data_dir)
> > +    if (!file.exists('input.txt')) {
> > +        download.file(url='https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/tinyshakespeare/input.txt',
> > +                      destfile='input.txt', method='wget')
> > +    }
> > +    setwd("..")
>
> I have changed it. :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/mxnet/pull/2329/files/ace088ea4cba974a18f13978009044c231eecead#r65903056>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABebVf58crQ7KFXmw8S7D6hX9YUXiXiWks5qJDMQgaJpZM4Iteaz>
> .
>



-- 
Qiang Kou
qkou@umail.iu.edu
School of Informatics and Computing, Indiana University